tests/transforms/test_optimization/test_reduce_non_clifford.py

```diff
@@ -0,0 +1,295 @@
+        ops = [qml.RZ(angle, wires=0)]
+
+        qs = QuantumScript(ops)
+        (new_qs,), _ = qml.transforms.zx.reduce_non_clifford(qs)
+
+        assert new_qs.operations == expected_ops
+
+    @pytest.mark.parametrize(
+        "measurements",
+        (
+            [],
+            [qml.expval(qml.Z(0))],
+            [qml.probs()],
+            [qml.state()],
+        ),
+    )
+    def test_transformed_tape(self, measurements):
+        """Test that the operations of the transformed tape match the expected operations
+        and that the original measurements are not touched."""
+        ops = [
+            qml.CNOT(wires=[0, 1]),
+            qml.T(wires=0),
+            qml.CNOT(wires=[3, 2]),
+            qml.T(wires=1),
+            qml.CNOT(wires=[1, 2]),
+            qml.T(wires=2),
+            qml.RZ(0.5, wires=1),
+            qml.CNOT(wires=[1, 2]),
+            qml.T(wires=1),
+            qml.CNOT(wires=[3, 2]),
+            qml.T(wires=0),
+            qml.CNOT(wires=[0, 1]),
+        ]
+        original_tape = qml.tape.QuantumScript(ops=ops, measurements=measurements)
+
+        (transformed_tape,), _ = qml.transforms.zx.reduce_non_clifford(original_tape)
+
+        expected_ops = [
+            qml.S(wires=0),
+            qml.CNOT(wires=[2, 3]),
+            qml.CNOT(wires=[0, 1]),
+            qml.RZ(2.070796326790258, wires=[1]),
+            qml.CNOT(wires=[1, 3]),
+            qml.T(wires=3),
+            qml.CNOT(wires=[1, 3]),
+            qml.CNOT(wires=[2, 3]),
+            qml.CNOT(wires=[0, 1]),
+        ]
+
+        assert transformed_tape.operations == expected_ops
+        assert transformed_tape.measurements == measurements
+
+    @pytest.mark.parametrize(
+        "params",
+        (
+            (0.0, 0.0),
+            (1.7, 0.0),
+            (0.0, -1.7),
+            (-3.2, 2.2),
+            (3.2, -2.2),
+        ),
+    )
+    def test_equivalent_state(self, params):
+        """Test that the output state returned by the transformed QNode matches
+        the output state returned by the original QNode up to a global phase."""
+        num_wires = 3
+        device = qml.device("default.qubit", wires=num_wires)
+
+        @qml.qnode(device)
+        def original_circ(x, y):
+            for i in range(num_wires):
+                qml.Hadamard(wires=i)
+            qml.T(wires=0)
+            qml.Hadamard(wires=0)
+            qml.Hadamard(wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.T(wires=0)
+            qml.RX(x, wires=1)
+            qml.S(wires=2)
+            qml.RX(y, wires=1)
+            qml.CNOT(wires=[1, 2])
+            return qml.state()
+
+        reduced_circ = qml.transforms.zx.reduce_non_clifford(original_circ)
+
+        state1 = original_circ(*params)
+        state2 = reduced_circ(*params)
+
+        # test that the states are equivalent up to a global phase
+        check = qml.math.fidelity_statevector(state1, state2)
+        assert np.isclose(check, 1)
```

Comment:
Could also compute the matrices and check their equality up to a global phase. Fine by me to leave it as is.